zk/stages/stage_sequence_execute_blocks.go

```diff
@@ -186,13 +186,29 @@ func finaliseBlock(
 
 	quit := batchContext.ctx.Done()
 	batchContext.sdb.eridb.OpenBatch(quit)
+
+	// if type 1 we need to do hash state
+	if batchContext.cfg.zk.UsingPMT() {
+		if err = NormalcyHashState(batchContext.s.LogPrefix(), batchContext.sdb.tx, batchContext.cfg.hashStateCfg, newHeader.Number.Uint64()-1, newHeader.Number.Uint64(), batchContext.ctx, log.New(batchContext.s.LogPrefix(), "trie", log.LvlTrace)); err != nil {
+			return nil, err
+		}
+	}
+
 	// this is actually the interhashes stage
-	newRoot, err := zkIncrementIntermediateHashes(batchContext.ctx, batchContext.s.LogPrefix(), batchContext.s, batchContext.sdb.tx, batchContext.sdb.eridb, batchContext.sdb.smt, newHeader.Number.Uint64()-1, newHeader.Number.Uint64())
+	var newRoot common.Hash
+	if batchContext.cfg.zk.UsingPMT() {
+		batchContext.cfg.intersCfg.checkRoot = false
```

Comment:
could we ensure that this is false for a sequencer rather than setting it to false here?